### PR TITLE
[WF-903] Integrate theming into Notification and Toast

### PR DESCRIPTION
**Pull request by ixTec**
# Integrate theming into Notification and Toast, tidy up some missed things 

### 🚀 Major Changes

- Integrate theming into **Notification** and **Toast**

### 🔧 Minor Changes

- Clean up styling for **Overlay**, **ScreenReaderOnly** and **ContentContainer**

### ❓ Misc Changes

- Fix missed styling refactors, jsdoc comments and other theming-related changes for an assortment of components

[WF-903]: https://datacamp.atlassian.net/browse/WF-903?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/dialog/__tests__/__snapshots__/dialog.spec.tsx.snap [hunk 1]
```diff
@@ -1664,7 +1664,7 @@ exports[`Dialog with decorative header as info variant 1`] = `
   left: 0;
   width: 100%;
   height: 100%;
-  color: #257DFE;
+  color: var(--wf-info--main, #5EB1FF);
```
[the remark this answers is not in the document]
Comment (dandenney) [reply src/dialog/__tests__/__snapshots__/dialog.spec.tsx.snap:1667]: But why do they also output a hex?

File: src/notification-card/styles.ts [hunk 2]
```diff
@@ -1,102 +1,60 @@
...
-  upgrade: {
-    color: tokens.colors.purpleLight,
-  },
-};
+const notificationStaticStyle = css({
+  position: 'relative',
+  display: 'flex',
+  borderRadius: tokens.borderRadius.medium,
+  backgroundColor: theme.background.contrast,
+  padding: '12px',
```
Comment (dandenney) [inline src/notification-card/styles.ts:14]: My semi-annual nudge for including things we use in the tokens or not having spacing tokens

File: src/notification-icon/styles.ts [hunk 3]
```diff
@@ -1,108 +1,89 @@
...
-  `;
+export function backgroundStyle({ variant }: BackgroundStyleOptions) {
+  // Creative way to get triangle shape for warning
+  return variant === 'warning'
+    ? css({
+        width: 0,
+        height: 0,
+        borderLeft: '6px solid transparent',
+        borderRight: '6px solid transparent',
+        borderBottom: `${ICON_SIZE} solid ${variantBackgroundMap[variant]}`,
```
Comment (dandenney) [inline src/notification-icon/styles.ts:81]: Is there a way to override `ICON_SIZE`? I see it's set to 12px, but if there is a way to pass in a value that changes that, I think we want the left and right to be `calc(${ICON_SIZE} / 2)`

File: src/notification-card/styles.ts [hunk 4]
```diff
@@ -1,102 +1,60 @@
...
-    bottom: -1px;
-    width: 4px;
-    background-color: ${variantMap[variant].color};
-    border-top-left-radius: ${tokens.borderRadius.medium};
-    border-bottom-left-radius: ${tokens.borderRadius.medium};
-  `;
+export function notificationStyle({ variant }: NotificationStyleStyleOptions) {
+  const borderColor = hexToRgba(
+    variant === 'warning' ? theme[variant].lighter : theme[variant].main,
+    0.6,
```
[the remark this answers is not in the document]
Comment (dandenney) [reply src/notification-card/styles.ts:24]: Ooh, I see we could use it as a token too

File: src/toast/styles.ts [hunk 5]
```diff
@@ -41,46 +53,39 @@ export function animatedWrapperStyle({
...
+  userSelect: 'none',
+  pointerEvents: 'all',
 
-    ${mediaQuery.aboveSmall} {
-      width: 360px;
-    }
-  `;
-}
+  [mediaQuery.aboveSmall]: {
+    width: '360px',
```
Comment (dandenney) [inline src/toast/styles.ts:78]: I wonder if we want this configurable?

File: src/notification-card/close-button.tsx [hunk 6]
```diff
@@ -20,22 +20,16 @@ const sizeMap = {
...
   size?: NonNullable<NotificationCardProps['size']>;
-  inverted?: boolean;
 } & Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'children'>;
 
-function CloseButton({
-  size = 'medium',
-  inverted = false,
-  ...restProps
-}: CloseButtonProps) {
+function CloseButton({ size = 'medium', ...restProps }: CloseButtonProps) {
```
[the remark this answers is not in the document]
Comment (ixTec) [reply src/notification-card/close-button.tsx:25]: One size internally, but customisable if they provide their own close button. A rare edge case that we support

File: src/notification-icon/styles.ts (hunk 3 above)
Comment (ixTec) [reply src/notification-icon/styles.ts:81]: No way to override the icon currently. Will update the icon size though

**Pull request merged**